File: 9_aula/Paragraph.cpp
#include <forward_list>
#include <list>
#include "PageElement.cpp"

class Paragraph : public PageElement {
private:
    std::list< std::shared_ptr<PageElement> > _mylist;
public:
    void addElement(std::shared_ptr<PageElement> e){
      _mylist.push_back(std::move(e));
    }

    std::string render() {
        std::string rendered = "<p>\n";
        for (std::shared_ptr<PageElement> e : _mylist)
            rendered += e->render() + "\n";
        return rendered + "</p>";
    }
};

File: 9_aula/Ver_2/Paragraph.java
import java.util.List;
import java.util.ArrayList;

public class Paragraph extends PageElement {

    private List<PageElement> _elements = new ArrayList<PageElement>();

    public void addElement(PageElement e) { _elements.add(e); }
    public void removeElement(PageElement e) { _elements.remove(e); }

    @Override
    public String render() {
        String rendered = "<p>\n";
        for (PageElement e : _elements)
            rendered += e.render() + "\n";
        return rendered + "</p>";
    }
}